src/auth.rs

```diff
@@ -108,31 +101,22 @@ impl Authenticator {
                                                       String,
                                                       u32);
 
-        let rnonce_b64 = match rnonce_opt {
-            Some(val) => val,
-            None => return Err(ResponseError(String::from("Invalid rnonce returned"))),
-        };
+        let rnonce_b64 = rnonce_opt
+            .ok_or(ResponseError(String::from("Invalid rnonce returned")))?;
```

Comment:
i think it's definitely a good thing to have, and we can incrementally build up a list of clippy rules we want hard style checks on; the only problem I have is keeping up with new or changing clippy rules, but I think the benefits outweigh the upkeep.